patches/gpu-config-gpu_finch_features.cc.patch

```diff
@@ -0,0 +1,12 @@
+diff --git a/gpu/config/gpu_finch_features.cc b/gpu/config/gpu_finch_features.cc
+index f05a624b2ba22c8a968bd3dd72a9ea870e431d82..c460b21791a1f6ce90966e5916afca5a0e1d36b9 100644
+--- a/gpu/config/gpu_finch_features.cc
++++ b/gpu/config/gpu_finch_features.cc
+@@ -323,6 +323,7 @@ bool IsUsingVulkan() {
+   if (IsDeviceBlocked(build_info->android_build_fp(),
+                       kVulkanBlockListByAndroidBuildFP.Get()))
+     return false;
++  BRAVE_VULKAN_BLOCK_LIST_DEVICE_CHECK
```

Comment:
it seems it is possible to get rid of this patch by overriding whole `IsUsingVulkan` and `IsDrDcEnabled`
```
//src/brave/chromium_src/gpu/config/gpu_finch_features.h
...
#define IsUsingVulkan IsUsingVulkan_ChromiumImpl
#define IsDrDcEnabled IsDrDcEnabled_ChromiumImpl

#include "src/gpu/config/gpu_finch_features.h"

#undef IsDrDcEnabled
#undef IsUsingVulkan

namespace features {
  
GPU_EXPORT bool IsUsingVulkan();
GPU_EXPORT bool IsDrDcEnabled();

}  // namespace features
...
```

```
//src/brave/chromium_src/gpu/config/gpu_finch_features.cc
...
#include "brave/chromium_src/gpu/config/gpu_finch_features.h"

#define IsUsingVulkan IsUsingVulkan_ChromiumImpl
#define IsDrDcEnabled IsDrDcEnabled_ChromiumImpl

#include "src/gpu/config/gpu_finch_features.cc"

#undef IsDrDcEnabled
#undef IsUsingVulkan

namespace features {

bool IsUsingVulkan() {
#if BUILDFLAG(IS_ANDROID)
   bool chromium_result = IsUsingVulkan_ChromiumImpl();
   if (!chromium_result) {
     return false;
   }
   
   const auto* build_info = base::android::BuildInfo::GetInstance();
   if (IsDeviceBlocked(build_info->board(), 
                       kVulkanBraveBlockListByBoard.Get())) {
     return false;
   }
   
   return true;
#else
  return base::FeatureList::IsEnabled(kVulkan);
#endif
}

bool IsDrDcEnabled() {
#if BUILDFLAG(IS_ANDROID)
  bool chromium_result = IsDrDcEnabled_ChromiumImpl();
  if (!chromium_result) {
    return false;
  }
  return IsUsingVulkan() ? base::FeatureList::IsEnabled(kEnableDrDcVulkan)
                         : true;
#else
  return false;
#endif
  
}
...
```
WDYT?

Reply:
@AlexeyBarabash  I looked on that solution as well, but decided to avoid it as basically our code becomes dependant on 2 functions. And what if `IsDrDcEnabled` is changed in upstream when they decide to not use `IsUsingVulkan` inside? We would simply miss that on rebase and everything will work not as expected. But I can do it that way if other reviewers think it's more reliable.

Reply:
that sounds good